Assets/Plugins/Editor/Uplift/Utils/GitHub.cs

```diff
@@ -86,27 +281,120 @@ private static GitHubAsset[] ExtractAssets(Dictionary<string, object> release)
+        {
+            X509Certificate[] githubCertificates = new X509Certificate[githubCertificatesString.Length];
+            for(int i = 0; i < githubCertificatesString.Length; i++)
+            {
+                githubCertificates[i] = new X509Certificate();
+                githubCertificates[i].Import(Encoding.ASCII.GetBytes(githubCertificatesString[i]));
+            }     
+
+            return githubCertificates;
+        }
+
+        public static bool CertificateValidationCallback(
+            System.Object sender,
+            X509Certificate certificate,
+            X509Chain chain,
+            SslPolicyErrors sslPolicyErrors
+        )
+        {
+            bool correct = true;
+            if(sslPolicyErrors == SslPolicyErrors.None) return true;       
+            
+            X509Certificate[] githubCertificates = GetCertificates();
+            
+            if(!(githubCertificates.Any(cert => cert.GetCertHashString() == certificate.GetCertHashString()) || UpliftPreferences.TrustUnknownCertificates()))
+            {
+                Debug.LogErrorFormat("The received certificate ({0}) is not known by Uplift. We cannot download the update package. You could update Uplift manually, or go to Preferences and set 'Trust unknown certificates' to true.", certificate.GetCertHashString());
+                Debug.Log("Known certificates are:");
+                foreach(X509Certificate cert in githubCertificates)
+                    Debug.Log(" -- " + cert.GetCertHashString());
+                return false;
+            }
+
+            for (int i=0; i < chain.ChainStatus.Length; i++) {
+                if (chain.ChainStatus[i].Status == X509ChainStatusFlags.RevocationStatusUnknown) {
+                    continue;
+                }
+                chain.ChainPolicy.RevocationFlag = X509RevocationFlag.EntireChain;
+                chain.ChainPolicy.RevocationMode = X509RevocationMode.Online;
+                chain.ChainPolicy.UrlRetrievalTimeout = new TimeSpan (0, 1, 0);
+                chain.ChainPolicy.VerificationFlags = X509VerificationFlags.AllFlags;
+                bool chainIsValid = chain.Build ((X509Certificate2)certificate);
+                if (!chainIsValid) {
+                    correct = false;
+                    break;
+                }
+            }
+
+            return correct;
+        }
+
+        public static Stream GetAssetStream(GitHubAsset asset)
+        {
+            return GetAssetStream(asset, "");
+        }
+        public static Stream GetAssetStream(GitHubAsset asset, string token)
+        {
+            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(asset.apiURL);
+            request.Method = "GET";
+            if(!string.IsNullOrEmpty(token))
+                request.Headers["Authorization"] = "token " + token;
+            request.Accept = "application/octet-stream";
+            request.UserAgent = "curl/7.47.0";
```

Comment:
We need to add our own agent.

I suggest

`Uplift GithubRepository/1.0 - https://github.com/DragonBox/uplift - some@email.for.us`

Reply:
We need an email :)

Reply:
Agreed for the email. I'll do that for the agent